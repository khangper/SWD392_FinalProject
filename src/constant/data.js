export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
