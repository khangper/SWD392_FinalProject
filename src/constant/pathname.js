export const PATH_NAME = {
  HOME: "/home",
  EXPLORE: "/explore",
  LIVE_OUTPUT: "/live_output/:id",
  ADD_LIVE_STREAM: "/add_livestream",
  ALL_LIVE_STREAM: "/all_livestream",
  ALL_INSTRUCTOR_LIST: "/allinstructorlist/:id",
  HELP: "/help",
  REPORT_HISTORY: "/reporthistory",
  SEND_FEEDBACK: "/sendfeedback",
  OTHER_INSTRUCTOR_VIEW: "/other-instructor-view/:id",
  MY_INSTRUCTOR_PROFILE: "/my-instructor-profile",
  SAVE_COURSES: "/save-courses",
  SETTING: "/setting",
  MY_COURSE: "/courses",
  LOGIN: "/login",
  FORGOT_PASSWORD: "/forgot-password",
  SIGN_UP: "/sign-up",
  SIGN_UP_STEP: "/SignUpStep",
  THANKS: "/thanks",
  ERROR_404: "/error404",
  INVOICE: "/invoice",
  PRESS: "/press",
  SEARCH_RESULT: "/search_result",
  PAID_MEMBERSHIP: "/paidmembership",
  ABOUT_US: "/about-us",
  OUR_BLOG: "/our-blog",
  BLOG_SINGLE_VIEW: "/blog-single-view",
  CHECKOUT: "/checkout",
  SHOPPING_CART: "/shoppingcart",
  HELP_VIEW_1: "/helpview1",
  HELP_VIEW_2: "/helpview2",
  COMPANY_DETAILS: "/company-details",
  CAREER: "/career",
  APPLY_JOB: "/apply-job",
  CONTACT_US: "/contactus",
  TERMS_OF_USE: "/termsofuse",
  CERTIFICATION_FILL_FORM: "/certificationfillform",
  CERTIFICATION_TEST: "/certificates-test",
  TEST_RESULT: "/test-result",
  MESSAGE: "/messages",
  COURSES_DETAIL_VIEW: "/courses-detail-view/:id",
  PREVIEWS: "/reviews",
  COMING_SOON: "/coming-soon",
  INS_DASHBOARD: "/ins-dashboard",
  PURCHASE_COURSE: "/PurchaseCourse",
  NOTIFICATON: "/notifications",
  DOWLOADCOURSE: "/dowloadcourse",
  CERTIFICATION_CENTER: "/certificationcenter",
  SITEMAP: "/sitemap",
  EARNING: "/earning",
  PAYOUT: "/payout",
  VERTIFICATION: "/vertification",
  MYCERTIFICATIONS: "/mycertifications",
  CREDIT: "/credits",
  STATEMENTIS: "/statementis",
  STATEMENTST: "/statementst",
  ANALYTICS: "/analytics",
  CREATE_NEW_COURSE: "/create-course",
  STD_DASHBOARD: "/student-dashboard",
  NOTIFICATIONSTUDENT: "/notificationsstudent",
};
