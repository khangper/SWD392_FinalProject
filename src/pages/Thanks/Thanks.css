/* Thankyou.css */
* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
}

.Thanks-wrapper {
    text-align: center;
    background-color: #121212;
    color: white;
    height: 100vh;
    margin-left: 0;
    margin-top: 0;
    padding: 0 !important;
}

.Thanks-row {
    max-width: 1200px;
    margin: 0 auto;
}

.Thanks-cmtk_group {
    width: 100%;
    padding: 100px 0;
}

.Thanks-ct-logo {
    padding-bottom: 80px;
    width: 100%;
    text-align: center;
}

.Thanks-ct-logo img {
    width: 150px;
}

.Thanks-cmtk_dt {
    /* padding-top: 100px; */
    text-align: center;
    padding-bottom: 68px;
}

.thnk_coming_title {
    font-size: 9em;
    font-weight: 600;
    font-family: 'Roboto', sans-serif;
    text-align: center;
    line-height: 1em;
    text-shadow: 1px 10px 6px rgba(0, 0, 0, 0.2);
    /* Ensure this is the correct property */
    padding-bottom: 29px;
}

.thnk_title1 {
    font-size: 24px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    text-align: center;
    line-height: 26px;
    margin-top: 0;
    text-shadow: 1px 10px 6px rgba(0, 0, 0, 0.2);
    /* Ensure this is the correct property */
}

.thnk_des {
    font-size: 16px;
    font-weight: 400;
    color: #fff;
    font-family: 'Roboto', sans-serif;
    text-align: center;
    line-height: 26px;
    text-shadow: 1px 10px 6px rgba(0, 0, 0, 0.2);
    /* Ensure this is the correct property */
    margin-top: 30px;
}

.thnk_des span {
    color: red;
}

.thnk_des a {
    color: red;
    text-decoration: none;
}

.thnk_des a:hover {
    text-decoration: underline;
}

.Thanks-tc_footer_main {
    display: flex;
    justify-content: space-between;
    align-items: center;
    font-size: 0.875rem;
    float: left;
    width: 100%;
    padding-top: 30px;
    padding-bottom: 30px;
    border-top: 1px solid #333;
    color: #fff;
}

.Thanks-tc_footer_left ul {
    list-style: none;
    padding: 0;
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
}

.Thanks-tc_footer_left ul li {
    margin: 0 10px;
}

.Thanks-tc_footer_left ul li a {
    color: #999;
    text-decoration: none;
}

.Thanks-tc_footer_left ul li a:hover {
    text-decoration: underline;
}

.Thanks-tc_footer_right p {
    color: #999;
}