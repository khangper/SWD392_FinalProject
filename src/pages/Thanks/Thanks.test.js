import React from "react";
import { renderWithProviders } from "../../constant/test-util";
import Thanks from "./Thanks";
import { screen } from "@testing-library/react";

describe("Thanks", () => {
    beforeAll(() => {
        jest.spyOn(console, "error").mockImplementation(() => { });
        jest.spyOn(console, "log").mockImplementation(() => { });
    });

    afterAll(() => {
        console.error.mockRestore();
        console.log.mockRestore();
    });
    it("renders the Thanks component", async () => {
        const { container } = renderWithProviders(<Thanks />);
        expect(container).toBeDefined();
        expect(container).toMatchSnapshot();
    });
})