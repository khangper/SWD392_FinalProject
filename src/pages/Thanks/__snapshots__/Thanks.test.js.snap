// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thanks renders the Thanks component 1`] = `
<div>
  <div
    class="Thanks-wrapper"
  >
    <div
      class="Thanks-row"
    >
      <div
        class="Thanks-col-md-12"
      >
        <div
          class="Thanks-cmtk_group"
        >
          <a
            href="/home"
          >
            <div
              class="Thanks-ct-logo"
            >
              <a
                href="index.html"
              >
                <img
                  alt=""
                  src="https://gambolthemes.net/html-items/cursus-new-demo/images/ct_logo.svg"
                />
              </a>
            </div>
          </a>
          <div
            class="Thanks-cmtk_dt"
          >
            <h1
              class="thnk_coming_title"
            >
              Thank You
            </h1>
            <h4
              class="thnk_title1"
            >
              Your Order is Confirmed!
            </h4>
            <p
              class="thnk_des"
            >
              Top Print Your Booking Order No.
              <span>
                 #ABE-ME-12345678
              </span>
               (Invoice) - 
              <a
                href="invoice.html"
              >
                Click Here
              </a>
            </p>
          </div>
          <div
            class="Thanks-tc_footer_main"
          >
            <div
              class="Thanks-tc_footer_left"
            >
              <ul>
                <li>
                  <a
                    href="about_us.html"
                  >
                    About
                  </a>
                </li>
                <li>
                  <a
                    href="press.html"
                  >
                    Press
                  </a>
                </li>
                <li>
                  <a
                    href="contact_us.html"
                  >
                    Contact Us
                  </a>
                </li>
                <li>
                  <a
                    href="coming_soon.html"
                  >
                    Advertise
                  </a>
                </li>
                <li>
                  <a
                    href="coming_soon.html"
                  >
                    Developers
                  </a>
                </li>
                <li>
                  <a
                    href="terms_of_use.html"
                  >
                    Copyright
                  </a>
                </li>
                <li>
                  <a
                    href="terms_of_use.html"
                  >
                    Privacy Policy
                  </a>
                </li>
                <li>
                  <a
                    href="terms_of_use.html"
                  >
                    Terms
                  </a>
                </li>
              </ul>
            </div>
            <div
              class="Thanks-tc_footer_right"
            >
              <p>
                Â© 2024 
                <strong>
                  Cursus
                </strong>
                . All Rights Reserved.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
